Basic demo

LukeSP opened the pull request:
creating pull request so Nigel can comment

LukeSP committed:
have added a basic specification demo without the extension. PLan is to have one with and one without. This is a first cut, and no where near completed.

LukeSP committed:
have added a basic specification demo without the extension. PLan is to have one with and one without. This is a first cut, and no where near completed.

LukeSP committed:
remove idea from repo

LukeSP committed:
have added a basic specification demo without the extension. PLan is to have one with and one without. This is a first cut, and no where near completed.

Commit:
added travis files
updated readMe files
added examples for with and without the extension
added settings file for multi project setup (while testing the extension and developing the demo untill the extension is published)

LukeSP committed:
trying to resolve travis issue with java home

LukeSP committed:
trying to resolve travis issue with java home

LukeSP committed:
added some missing files to git

LukeSP committed:
travis with only java8 env to test

LukeSP committed:
travis with pre install permissions setting

LukeSP committed:
removed *.jar files form git ignore for travis build

LukeSP committed:
added gradle wrappers

nigelcharman commented on .travis.yml:
```diff
@@ -0,0 +1,18 @@
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
```
On success publish to github pages. See https://github.com/concordion/concordion-screenshot-extension-demo/blob/master/.travis.yml

nigelcharman commented on README.md:
```diff
@@ -1 +1,18 @@
+   Import as a Gradle project. This may require additional plugins to be installed to support Gradle.
+   You can then run the Gradle task 'check' under verification which will run the tests.
+   The output can be found here:
+   \Users\<username>\AppData\Local\Temp\concordion\org\concordion\ext\specification
```
See wording at https://github.com/concordion/concordion-screenshot-extension-demo

nigelcharman commented on build.gradle:
```diff
@@ -0,0 +1,22 @@
+}
+
+dependencies {
+    testCompile 'org.concordion:concordion:2.1.1'
```
2.1.2

nigelcharman commented on src/test/java/org/concordion/ext/specification/withExpectedToFailInfoExtension.java:
```diff
@@ -0,0 +1,28 @@
+package org.concordion.ext.specification;
```
Add example using `@Extension` with `setStyle()` method

LukeSP committed:
updated demo travis file based on changes to extension travis.
added over-writing feature to some fields in the extension
added a test to test and demonstrate the overwriting

LukeSP committed:
Removed unnecesary packages and files
updated build files
updated Readme files

nigelcharman committed:
Updated gitignore

nigelcharman committed:
Updated gradle wrapper

LukeSP committed:
renamed files to have upper case first character

LukeSP committed:
renamed files to have upper case first character

LukeSP committed:
added index file

LukeSP committed:
Note added for for issue where multiple listeners are created when both @Extension and @Extensions are in/on a class
updated some
updated some syntax
added in some placeholders for further enhancements around other concordion statuses

LukeSP committed:
added run command in index file

LukeSP committed:
gave a more informative title to the index page

LukeSP committed:
Should not apply the @Extensions annotation on the class if an @Extension call is made to override the default behaviour in the said class

LukeSP committed:
added index class

LukeSP committed:
Update README.md

Commit:
updated gradle files

Commit:
commit gradlew change

Commit:
gitignores build dir

Commit:
gradlew properties

Commit:
removed .gradlefiles from repo

Commit:
import the extension
gradle dependency line
update gradle settings project name
update tests:
- new extension name
- ignored and unimplemented tests
- updated spec names

Commit:
readme cleanup
